src/components/molecules/OverlayLoader.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {FC } from 'react';
+import Loader from '../atoms/Loader';
+import Box from '@material-ui/core/Box';
+import { makeStyles } from '@material-ui/core';
+import { borderColor } from '../../style/_globals';
+import { useStore } from '../../store/storeConfig';
+import RootStore from '../../store/root.store';
+import { toJS } from 'mobx';
+import { observer } from 'mobx-react-lite';
+
+
+const useStyles = makeStyles({
+   over: {
+    minWidth: '100%',
+    minHeight: '100%',
+    backgroundColor: borderColor,
+    opacity: 0.1,
+    zIndex: 2,
+    display: 'flex',
+    flexDirection: 'column'
+   },
+   loader: {
+       alignSelf: 'center',
+       marginTop: '10px'
+   }
+})
+
+interface IProps {
+    dataToLoad: string 
+}
+const OverlayLoader: FC<IProps> = ({ dataToLoad, children}) => {
```

Comment:
why sending dataToLoad?
This is a mere loader - although it is layout - specific 
The only property it need's is a boolean flag